ui/components/src/model/legend.ts

```diff
@@ -24,15 +24,23 @@ export interface LegendOptions {
 export interface LegendItem {
   id: string;
   label: string;
-  isSelected: boolean;
   color: string;
-  onClick: MouseEventHandler<HTMLElement>;
+  onClick?: MouseEventHandler<HTMLElement>;
 }
 
 export type LegendPositionConfig = {
   label: string;
 };
 
+/**
+ * State of selected items in the legend.
+ * - When "ALL", all legend items are selected, but not visually highlighted.
+ * - Otherwise, it is a Record that associates legend item ids with a boolean
+ *   value. When the associated entry for a legend item is `true`, that item
+ *   will be treated as selected and visually highlighted.
+ */
+export type SelectedLegendItemState = Record<LegendItem['id'], boolean> | 'ALL';
```

Comment:
I wouldn't be opposed to calling this `SelectedLegendItems` type, but I think this is fine too!